exp/controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -206,10 +208,15 @@ func (r *AzureManagedControlPlaneReconciler) reconcileNormal(ctx context.Context
 		return reconcile.Result{}, err
 	}
 
-	if err := newAzureManagedControlPlaneReconciler(scope).Reconcile(ctx, scope); err != nil {
+	if err := newAzureManagedControlPlaneReconciler(scope).Reconcile(ctx); err != nil {
 		return reconcile.Result{}, errors.Wrapf(err, "error creating AzureManagedControlPlane %s/%s", scope.ControlPlane.Namespace, scope.ControlPlane.Name)
 	}
 
+	// Reconcile kubeconfig
+	if err := r.reconcileKubeConfig(ctx, scope); err != nil {
```

Comment:
nit: why move this out of the other file?

Reply:
I have to check it tomorrow when I get to my computer, but I believe it was because some vars required by this func were removed/moved from the reconciler.

Reply:
Took a look now, so scope is not passed around in reconciler (`azuremanagedcontrolplane_reconciler.go`) funcs as a parameter, but I guess I could add it as a struct field. That would reduce the size of the PR slightly. I'll do that.

Reply:
I moved back this function back to reconciler file, although it looks a bit different now due to overall refactoring.